addons/mrp/stock.py

```diff
@@ -191,7 +191,7 @@ def action_consume(self, cr, uid, ids, product_qty, location_id=False, restrict_
             # Compare with numbers of move uom as we want to avoid a split with 0 qty
             quantity_rest_uom = move.product_uom_qty - self.pool.get("product.uom")._compute_qty_obj(cr, uid, move.product_id.uom_id, product_qty, move.product_uom)
             if float_compare(quantity_rest_uom, 0, precision_rounding=move.product_uom.rounding) != 0:
-                new_mov = self.split(cr, uid, move, quantity_rest, context=context)
+                new_mov = self.split(cr, uid, move, quantity_rest, restrict_lot_id=restrict_lot_id, context=context)
```

Comment:
It is "move" that is set to done, so there the restrict_lot_id is important.  In the case of the new move, it is still uncertain which lot you will use.